### PR TITLE
neo4j version update query in aws

### PR DESCRIPTION
**Pull request by sagarnavale**

### CODE REVIEW

File: cartography/data/jobs/cleanup/aws_account_dns_cleanup.json [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {
   "statements": [
-  {
-    "query": "MATCH (n:AWSDNSRecord)-[:MEMBER_OF_DNS_ZONE]->(:AWSDNSZone)<-[:RESOURCE]-(:AWSAccount{id: {AWS_ID}})<-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}}) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
-    "iterative": true,
-    "iterationsize": 100
-  }],
+    {
+      "query": "MATCH (n:AWSDNSRecord)-[:MEMBER_OF_DNS_ZONE]->(:AWSDNSZone)<-[:RESOURCE]-(:AWSAccount{id: $AWS_ID})<-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}}) WHERE n.lastupdated <> $UPDATE_TAG WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (ganeshjagdale9545) [inline cartography/data/jobs/cleanup/aws_account_dns_cleanup.json:4]: make change for `WORKSPACE_ID`

File: cartography/data/jobs/cleanup/aws_import_apigateway_cleanup.json [hunk 2]
```diff
@@ -1,45 +1,45 @@
...
+      "query": "MATCH (n:APIGatewayStage)<-[:ASSOCIATED_WITH]-(:APIGatewayRestAPI)<-[:RESOURCE]-(:AWSAccount{id: $AWS_ID})<-[:OWNER]-(:CloudanixWorkspace{id: $WORKSPACE_ID}) WHERE n.lastupdated <> $UPDATE_TAG WITH n LIMIT $LIMIT_SIZE DETACH DELETE (n) return COUNT(*) as TotalCompleted",
       "iterative": true,
       "iterationsize": 100
     },
     {
-      "query": "MATCH (:APIGatewayStage)<-[r:ASSOCIATED_WITH]-(:APIGatewayRestAPI)<-[:RESOURCE]-(:AWSAccount{id: {AWS_ID}})<-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}}) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
+      "query": "MATCH (:APIGatewayStage)<-[r:ASSOCIATED_WITH]-(:APIGatewayRestAPI)<-[:RESOURCE]-(:AWSAccount{id: $AWS_ID})<-[:OWNER]-(:CloudanixWorkspace{id: $WORKSPACE_ID}) WHERE r.lastupdated <> $UPDATE_TAG WITH r LIMIT$LIMIT_SIZE DELETE (r) return COUNT(*) as TotalCompleted",
```
Comment (ganeshjagdale9545) [inline cartography/data/jobs/cleanup/aws_import_apigateway_cleanup.json:19]: add space between `LIMIT` and `$LIMIT_SIZE`

File: cartography/intel/aws/cloudwatch.py [hunk 3]
```diff
@@ -47,19 +47,19 @@ def load_event_buses(session: neo4j.Session, event_buses: List[Dict], current_aw
...
     ON CREATE SET bus.firstseen = timestamp(),
         bus.arn = record.Arn
-    SET bus.lastupdated = {aws_update_tag},
+    SET bus.lastupdated = $aws_update_tag,
         bus.name = record.Name,
         bus.region = record.region,
         bus.policy = record.Policy
     WITH bus
-    MATCH (owner:AWSAccount{id: {AWS_ACCOUNT_ID}})
+    MATCH (owner:AWSAccount{id: $AWS_ACCOUNT_ID}})
```
Comment (ganeshjagdale9545) [inline cartography/intel/aws/cloudwatch.py:59]: remove extra `}`

File: cartography/intel/aws/cloudwatch.py [hunk 4]
```diff
@@ -310,10 +310,10 @@ def _load_cloudwatch_alarm_tx(tx: neo4j.Transaction, alarms: List[Dict], current
         alarm.actions_enabled = record.ActionsEnabled,
         alarm.metric_name = record.MetricName
     WITH alarm
-    MATCH (owner:AWSAccount{id: {AWS_ACCOUNT_ID}})
+    MATCH (owner:AWSAccount{id: $AWS_ACCOUNT_ID})
     MERGE (owner)-[r:RESOURCE]->(alarm)
     ON CREATE SET r.firstseen = timestamp()
-    SET r.lastupdated = {aws_update_tag}
+    SET r.lastupdated = aws_update_tag
```
Comment (ganeshjagdale9545) [inline cartography/intel/aws/cloudwatch.py:316]: `$` missing here

**Review by ganeshjagdale9545 — CHANGES_REQUESTED** [review]
few minor changes needed also make changes in `cartography/data/jobs/indexes.cypher`

File: cartography/data/jobs/cleanup/aws_import_apigateway_cleanup.json (hunk 2 above)
Comment (ganeshjagdale9545) [reply cartography/data/jobs/cleanup/aws_import_apigateway_cleanup.json:19]: make this change

File: cartography/data/jobs/cleanup/azure_database_account_cleanup.json [hunk 5]
```diff
@@ -1,85 +1,85 @@
 {
   "statements": [
     {
-      "query": "MATCH (n:AzureCosmosDBLocation)<-[:WRITE_PERMISSIONS_FROM]-(:AzureDatabaseAccount)<-[:RESOURCE]-(:AzureSubscription{id: {AZURE_SUBSCRIPTION_ID}})<-[:RESOURCE]-(:AzureTenant{id: {AZURE_TENANT_ID}})<-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}}) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
+      "query": "MATCH (n:AzureCosmosDBLocation)<-[:WRITE_PERMISSIONS_FROM]-(:AzureDatabaseAccount)<-[:RESOURCE]-(:AzureSubscription{id: $AZURE_SUBSCRIPTION_ID})<-[:RESOURCE]-(:AzureTenant{id: AZURE_TENANT_ID})<-[:OWNER]-(:CloudanixWorkspace{id: $WORKSPACE_ID}) WHERE n.lastupdated <> $UPDATE_TAG WITH n LIMIT $LIMIT_SIZE DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (ganeshjagdale9545) [inline cartography/data/jobs/cleanup/azure_database_account_cleanup.json:4]: `$` missing here

File: cartography/data/jobs/cleanup/azure_database_account_cleanup.json [hunk 6]
```diff
@@ -1,85 +1,85 @@
...
       "iterative": true,
       "iterationsize": 100
     },
     {
-      "query": "MATCH (:AzureCosmosDBLocation)<-[r:WRITE_PERMISSIONS_FROM]-(:AzureDatabaseAccount)<-[:RESOURCE]-(:AzureSubscription{id: {AZURE_SUBSCRIPTION_ID}})<-[:RESOURCE]-(:AzureTenant{id: {AZURE_TENANT_ID}})<-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}}) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
+      "query": "MATCH (:AzureCosmosDBLocation)<-[r:WRITE_PERMISSIONS_FROM]-(:AzureDatabaseAccount)<-[:RESOURCE]-(:AzureSubscription{id: {AZURE_SUBSCRIPTION_ID}})<-[:RESOURCE]-(:AzureTenant{id: AZURE_TENANT_ID})<-[:OWNER]-(:CloudanixWorkspace{id: $WORKSPACE_ID}) WHERE r.lastupdated <> $UPDATE_TAG WITH r LIMIT $LIMIT_SIZE DELETE (r) return COUNT(*) as TotalCompleted",
```
Comment (ganeshjagdale9545) [inline cartography/data/jobs/cleanup/azure_database_account_cleanup.json:9]: `$ `missing here

File: cartography/data/jobs/cleanup/azure_import_virtual_machines_scale_sets_extensions_cleanup.json [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
             "iterative": true,
             "iterationsize": 100
         },
         {
-            "query": "MATCH (:AzureVirtualMachineScaleSetExtension)<-[r:CONTAIN]-(:AzureVirtualMachineScaleSet)<-[:RESOURCE]-(:AzureSubscription{id: {AZURE_SUBSCRIPTION_ID}})<-[:RESOURCE]-(:AzureTenant{id: {AZURE_TENANT_ID}})<-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}}) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
+            "query": "MATCH (:AzureVirtualMachineScaleSetExtension)<-[r:CONTAIN]-(:AzureVirtualMachineScaleSet)<-[:RESOURCE]-(:AzureSubscription{id: $AZURE_SUBSCRIPTION_ID)<-[:RESOURCE]-(:AzureTenant{id: $AZURE_TENANT_ID})<-[:OWNER]-(:CloudanixWorkspace{id: $WORKSPACE_ID}) WHERE r.lastupdated <> $UPDATE_TAG WITH r LIMIT $LIMIT_SIZE DELETE (r) return COUNT(*) as TotalCompleted",
```
Comment (ganeshjagdale9545) [inline cartography/data/jobs/cleanup/azure_import_virtual_machines_scale_sets_extensions_cleanup.json:9]: `}` missing

File: cartography/data/jobs/cleanup/gcp_kms_cleanup.json [hunk 8]
```diff
@@ -1,41 +1,41 @@
 {
 	"statements": [
 		{
-			"query": "MATCH (n:GCPKMSCryptoKey)<-[:RESOURCE]-(:GCPKMSKeyRing)<-[:RESOURCE]-(:GCPLocation)<-[:RESOURCE]-(:GCPProject{id: {GCP_PROJECT_ID}})<-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}}) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
+			"query": "MATCH (n:GCPKMSCryptoKey)<-[:RESOURCE]-(:GCPKMSKeyRing)<-[:RESOURCE]-(:GCPLocation)<-[:RESOURCE]-(:GCPProject{id: $GCP_PROJECT_ID})<-[:OWNER]-(:CloudanixWorkspace{id: $WORKSPACE_ID}) WHERE n.lastupdated <> $UPDATE_TAG WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (ganeshjagdale9545) [inline cartography/data/jobs/cleanup/gcp_kms_cleanup.json:4]: make change for `LIMIT_SIZE `

File: cartography/data/jobs/cleanup/gcp_kms_cleanup.json [hunk 9]
```diff
@@ -1,41 +1,41 @@
...
 			"iterative": true,
 			"iterationsize": 100,
 			"__comment__": "Delete GCP KMS Cryptokeys that no longer exist and detach them from all previously connected nodes."
 		},
 		{
-			"query": "MATCH (:GCPKMSCryptoKey)<-[r:RESOURCE]-(:GCPKMSKeyRing)<-[:RESOURCE]-(:GCPLocation)<-[:RESOURCE]-(:GCPProject{id: {GCP_PROJECT_ID}})<-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}}) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
+			"query": "MATCH (:GCPKMSCryptoKey)<-[r:RESOURCE]-(:GCPKMSKeyRing)<-[:RESOURCE]-(:GCPLocation)<-[:RESOURCE]-(:GCPProject{id: $GCP_PROJECT_ID})<-[:OWNER]-(:CloudanixWorkspace{id: $WORKSPACE_ID}) WHERE r.lastupdated <> $UPDATE_TAG WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
```
Comment (ganeshjagdale9545) [inline cartography/data/jobs/cleanup/gcp_kms_cleanup.json:10]: same as above

File: cartography/data/jobs/cleanup/gcp_kms_cleanup.json [hunk 10]
```diff
@@ -1,41 +1,41 @@
...
 			"iterative": true,
 			"iterationsize": 100,
 			"__comment__": "Remove GCP GCPKMSCryptoKey-to-Keyrings relationships that are out of date."
 		},
 		{
-			"query": "MATCH (n:GCPKMSKeyRing)<-[:RESOURCE]-(:GCPLocation)<-[:RESOURCE]-(:GCPProject{id: {GCP_PROJECT_ID}})<-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}}) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
+			"query": "MATCH (n:GCPKMSKeyRing)<-[:RESOURCE]-(:GCPLocation)<-[:RESOURCE]-(:GCPProject{id: $GCP_PROJECT_ID})<-[:OWNER]-(:CloudanixWorkspace{id: $WORKSPACE_ID}) WHERE n.lastupdated <> $UPDATE_TAG WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (ganeshjagdale9545) [inline cartography/data/jobs/cleanup/gcp_kms_cleanup.json:16]: same as above

**Review by ganeshjagdale9545 — CHANGES_REQUESTED** [review]
few minor changes needed

**Review by ganeshjagdale9545 — APPROVED** [review]
looks good

**Comment by ganeshjagdale9545** [comment]
@sagarnavale can you resolve the conflicts

**Review by ganeshjagdale9545 — APPROVED** [review]
looks good

**Comment by mpurusottamc** [comment]
looks good.

**Pull request merged**